### PR TITLE
IOU Participant page search validation

### PR DESCRIPTION
**Pull request by Santhosh-Sellavel**
### Details
Added search validation for IOU Participant Pages also like as it is in New Chat/Group search pages

### Fixed Issues

$ https://github.com/Expensify/App/issues/4690

### Tests & QA Steps
1. Click ➕  FAB icon.
2. Select New IOU Request/Split options
3. Enter invalid search input 

### Fix
A validation message will show for invalid inputs on IOU participant pages search also.

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web
Screenshot 2021-09-15 at 12 47 58 AM

#### Mobile Web
Simulator Screen Shot - iPhone 11 - 2021-09-15 at 00 50 03

#### Desktop
Screenshot 2021-09-15 at 12 48 29 AM

#### iOS
Simulator Screen Shot - iPhone 12 - 2021-09-15 at 00 52 13

#### Android
Screenshot_1631647362

### CODE REVIEW

File: src/libs/OptionsListUtils.js
Comment (Santhosh-Sellavel) [inline src/libs/OptionsListUtils.js:722]: @tgolen
While using this method for the split I found a case where I didn't give any search input just picked users from the recent list. Since users are picked are excluded from `personalDetails`, the `personalDetails` list become empty that `hasSelectableOptions` is false so `No result found` is returned. So I added a check that `searchValue` should be available with the existing checks.

This additionally added along proceed with proposed changes. If you are good with this change, I'll test it once again and mark it ready for review!

Comment (tgolen) [reply src/libs/OptionsListUtils.js:722]: I think that's fine, yeah. Thank you for the heads up.

Comment (tgolen) [reply src/libs/OptionsListUtils.js:722]: Would you mind adding a comment to the code to explain why `searchValue` needs to be in that logic? I just don't want someone else to come along and see it and think it's OK to remove it without understanding the implications of it being there.

**Review by tgolen — DISMISSED** [review]

File: src/libs/OptionsListUtils.js
Comment (Santhosh-Sellavel) [reply src/libs/OptionsListUtils.js:722]: In short I think -> `Without a searchValue should not return any search validation messages`
Is this is okay?
Do you have anything specific comment in mind @tgolen?

**Comment by kadiealexander** [comment]
Hi @Santhosh-Sellavel, I've placed this issue on hold as per [this update](https://expensify.slack.com/archives/C01GTK53T8Q/p1631754482374800), as we are prioritising issues related to a feature release scheduled for 9/24. As an apology for the delay, we will add a $100 bonus as a thank you for waiting.

File: src/libs/OptionsListUtils.js
Comment (tgolen) [reply src/libs/OptionsListUtils.js:722]: Yeah, that's pretty good! I would expand it a little further like this:

> Without a search value, it would be very confusing to see a search validation message. Therefore, this skips the validation when there is no search value.

Comment (Santhosh-Sellavel) [reply src/libs/OptionsListUtils.js:724]: > Without a search value, it would be very confusing to see a search validation message. Therefore, this skips the validation when there is no search value.

This is better, updated the comment!

**Review by tgolen — DISMISSED** [review]
Looks great, thanks! I'm approving, but won't merge right now since it's on HOLD.

**Review by parasharrajat — CHANGES_REQUESTED** [review]
Apart from the comments below, another issue that I have found is when the maximum number of users is selected (and we hide the contact list), now if you search for something and the user is not in the hidden list, then an error is shown which is bad.
image

File: src/pages/iou/steps/IOUParticipantsPage/IOUParticipantsRequest.js
Comment (Santhosh-Sellavel) [reply src/pages/iou/steps/IOUParticipantsPage/IOUParticipantsRequest.js:119]: can you add your video here?

Comment (Santhosh-Sellavel) [reply src/pages/iou/steps/IOUParticipantsPage/IOUParticipantsRequest.js:119]: > the List has a Recent chats section if you search for something that is shown in recent chats

Yes, that's the expected behavior.

**Comment by Santhosh-Sellavel** [comment]
> @Santhosh-Sellavel Video.
> 
>  output_file.mp4

Ok, I get it now.

File: src/pages/iou/steps/IOUParticipantsPage/IOUParticipantsSplit.js
Comment (Santhosh-Sellavel) [reply src/pages/iou/steps/IOUParticipantsPage/IOUParticipantsSplit.js:202]: > When you see the message that you have selected the maximum number of participants, we hide the selection list (contains recent chats and contacts section) which is necessary as we don't want the user to select more users.

Cool, we are on the same page here.

In this case, if you search for a user who does not exist in the selection list(which should be hidden when maximum users are selected), No result found is shown which is the issue.

Only if the user is not hidden in the list, we show `no results found`.

@parasharrajat

Comment (Santhosh-Sellavel) [reply src/pages/iou/steps/IOUParticipantsPage/IOUParticipantsSplit.js:202]: @parasharrajat Video for reference

Uploading NoResults.mov…

Comment (parasharrajat) [reply src/pages/iou/steps/IOUParticipantsPage/IOUParticipantsSplit.js:202]: I think you didn't get it. Let me say this in a different way.

When the list is hidden and you are searching , you should not show any error has search has no effect on the list of contacts shown below. 

Having a `No result found` message puts a wrong meaning when the list below is not affected by the search.

Comment (Santhosh-Sellavel) [reply src/pages/iou/steps/IOUParticipantsPage/IOUParticipantsSplit.js:202]: Okay

File: src/pages/iou/steps/IOUParticipantsPage/IOUParticipantsSplit.js
Comment (parasharrajat) [inline src/pages/iou/steps/IOUParticipantsPage/IOUParticipantsSplit.js:201]: it is not readable. I would suggest swapping the condition so that condition would be `!maxParticipantsReached`.

**Review by parasharrajat — CHANGES_REQUESTED** [review]
One minor change and otherwise looks good now.

File: src/pages/iou/steps/IOUParticipantsPage/IOUParticipantsSplit.js
Comment (Santhosh-Sellavel) [reply src/pages/iou/steps/IOUParticipantsPage/IOUParticipantsSplit.js]: Done

**Review by parasharrajat — APPROVED** [review]
LGTM. Tests well. Ready for final review after n6-hold is lifted.

**Review by marcaaron — APPROVED** [review]

**Pull request merged**
